IntelliTect.Analyzer/IntelliTect.Analyzer.Test/UnusedLocalVariableTests.cs

```diff
@@ -140,10 +140,82 @@ public void Descriptor_ContainsExpectedValues()
             Assert.AreEqual("Flow", diagnostic.Category);
             Assert.AreEqual(DiagnosticSeverity.Info, diagnostic.DefaultSeverity);
             Assert.AreEqual(true, diagnostic.IsEnabledByDefault);
-            Assert.AreEqual("All local variables should be accessed", diagnostic.Description);
+            Assert.AreEqual("All local variables should be accessed, or named with underscores to indicate they are unused", diagnostic.Description);
             Assert.AreEqual("https://github.com/IntelliTect/CodingStandards", diagnostic.HelpLinkUri);
         }
 
+        [TestMethod]
+        public void LambdaExpressionWithDiscard_NoDiagnosticInformationReturned()
+        {
+            string test = @"
+using System;
+
+namespace ConsoleApplication1
+{
+    class TypeName
+    {   
+        public void Foo()
+        {
+            Bar(_ => { return true; });
+            bool Bar(Func<bool, bool> func)
+            {
+                return func(true);
+            }
+        }
+    }
+}";
+
+            VerifyCSharpDiagnostic(test);
+        }
+
+        [TestMethod]
+        public void LambdaExpressionWithTwoDiscards_NoDiagnosticInformationReturned()
+        {
+            string test = @"
+using System;
+
+namespace ConsoleApplication1
+{
+    class TypeName
+    {   
+        public void Foo()
+        {
+            Bar(__ => { return true; });
+            bool Bar(Func<bool, bool> func)
+            {
+                return func(true);
+            }
+        }
+    }
+}";
+
+            VerifyCSharpDiagnostic(test);
+        }
+
+        [TestMethod]
+        public void LambdaMethodWithDiscard_NoDiagnosticInformationReturned()
```

Comment:
Isn't this the same as the first test? Should this be testing the opposite case confirming the un-used lambda parameters are flagged

Reply:
I honestly wasn't sure, so figured I'd err on the side of caution. :) If having or not having braces is fundamentally the same, I can remove the test.

But, yes, I did forget to add tests for the opposite case. I will do that now!